src/test/java/org/ays/auth/tests/RoleCreateTest.java

```diff
@@ -0,0 +1,153 @@
+package org.ays.auth.tests;
+
+import io.restassured.response.Response;
+import org.ays.auth.datasource.PermissionDataSource;
+import org.ays.auth.datasource.RoleDataSource;
+import org.ays.auth.endpoints.AuthEndpoints;
+import org.ays.auth.endpoints.RoleEndpoints;
+import org.ays.auth.payload.LoginPayload;
+import org.ays.auth.payload.RoleCreatePayload;
+import org.ays.common.model.enums.AysErrorMessage;
+import org.ays.common.util.AysConfigurationProperty;
+import org.ays.common.util.AysDataProvider;
+import org.ays.common.util.AysResponseSpecs;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class RoleCreateTest {
+
+    private String accessToken;
+
+    @BeforeMethod
+    private void setUpToken() {
+
+        LoginPayload loginPayload = LoginPayload.generateAsTestVolunteerFoundationAdmin();
+        accessToken = loginAndGetAccessToken(loginPayload);
+    }
```

Comment:
```suggestion
class RoleCreateTest {

    private final String accessToken;

    @BeforeMethod
    private void setUpToken() {
        LoginPayload loginPayload = LoginPayload.generateAsTestVolunteerFoundationAdmin();
        this.accessToken = this.loginAndGetAccessToken(loginPayload);
    }
```

Reply:
@agitrubard class'i public yapmamalı miyim ?

Reply:
@htczrrsz 

1. Eğer ilgili sınıf başka bir paketteki bir sınıftan çağrılmıyor ise `package-private` olması her zaman daha iyi bir yaklaşımdır. `package-private` yaptığımızda bir hataya yol açmazsa bu şekilde ilerleyebiliriz ✅ 
2. Aslında normal şartlarda olması beklenen fakat metot olduğu için doğru `final` keyword'ünü ekleyemeyiz. `final`'ı eklememizin sebebi oluşturulmasına zorlamak ama sorun yok burada final eklemeyebiliriz. Yorumumu buna göre düzenledim ✅ 
3. İlgili değişkenin/metodun bu sınıf içerisinde tanımlanmış bir değişken/metot olduğunu vurgulamak amacıyla da kullanılması gerektiğini düşünüyorum. `this.` ile başladığında bu sınıf içerisinde olduğunu gözle görebilmemizi kolaylaştırıyor. Kodu ne kadar hızlı okuyup anlayabiliyorsak, o kadar iyi olduğunu söyleyebiliyoruz. Bu sebeple her zaman static olmayan metotlarda eklemeyi daha doğru buluyorum.

Reply:
@agitrubard 
package-private oldugunda test run butonları kalkıyor , o yüzden bu duruma gore düzenliyorum